Assets/LeapMotion/Core/Scripts/XR/XRHeightOffset.cs

```diff
@@ -0,0 +1,129 @@
+/******************************************************************************
+ * Copyright (C) Leap Motion, Inc. 2011-2017.                                 *
+ * Leap Motion proprietary and  confidential.                                 *
+ *                                                                            *
+ * Use subject to the terms of the Leap Motion SDK Agreement available at     *
+ * https://developer.leapmotion.com/sdk_agreement, or another agreement       *
+ * between Leap Motion and you, your company or other organization.           *
+ ******************************************************************************/
+
+using UnityEngine;
+using System;
+using System.Linq;
+using Leap.Unity.Attributes;
+
+public class XRHeightOffset : MonoBehaviour {
+
+  #region Inspector
+
+  [Header("Height Offset When XR Space Tracking Type Is Stationary")]
+
+  [SerializeField]
+  [OnEditorChange("stationaryHeightOffset")]
+  [Tooltip("This offset only used when the XR space tracking type is 'Stationary.' This "
+         + "is the tracking type for Oculus setups with a single tracking camera and "
+         + "for Android VR devices without position tracking. Two-camera Oculus setups "
+         + "and Vive headsets, for example, do not require a height offset. They center "
+         + "the room-scale floor on the Rig transform, and this script will not apply "
+         + "a height offset when the RoomScale space-tracking type is detected.")]
+  private float _stationaryHeightOffset = 1.6f; // average human height (or so)
+  public float stationaryHeightOffset {
+    get { return _stationaryHeightOffset; }
+    set {
+      _stationaryHeightOffset = value;
+
+      applyHeightOffset();
+    }
+  }
+
+  #region Auto Recenter
+
+  [Header("Auto Recenter")]
+
+  [SerializeField]
+  [Tooltip("If the detected XR device is present and supports userPresence, "
+         + "checking this option will detect when the user puts on the device headset "
+         + "and call InputTracking.Recenter.")]
+  private bool autoRecenterOnUserPresence = true;
+
+  private UnityEngine.XR.UserPresenceState _lastUserPresence;
+
+  #endregion
+
+  #region Runtime Height Adjustment
+
+  [Header("Runtime Height Adjustment")]
+
+  public bool enableRuntimeAdjustment = true;
+
+  [DisableIf("enableRuntimeAdjustment", isEqualTo: false)]
+  [Tooltip("Press this key on the keyboard to adjust the height offset up by stepSize.")]
+  public KeyCode stepUpKey = KeyCode.UpArrow;
+
+  [DisableIf("enableRuntimeAdjustment", isEqualTo: false)]
+  [Tooltip("Press this key on the keyboard to adjust the height offset down by stepSize.")]
+  public KeyCode stepDownKey = KeyCode.DownArrow;
+
+  [DisableIf("enableRuntimeAdjustment", isEqualTo: false)]
+  public float stepSize = 0.1f;
+
+  #endregion
+
+  #endregion
+
+  #region Unity Events
+
+  private Vector3 _startingPosition = Vector3.zero;
+
+  private void Start() {
+    _startingPosition = this.transform.position;
```

Comment:
Using a starting position would prevent the transform from being moved normally correct?  If this is desired behavior, we might want to consider enforcing it every frame, because otherwise people might be bitten when their transform moves back to a location whenever the user takes off their headset and puts it back on again.  (or adjusts the stationaryHeightOffset)